gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/artifacts/ExtractSingleFileOrManifest.java

```diff
@@ -57,11 +57,11 @@ public final void transform(TransformOutputs outputs) {
                 return;
             }
 
-            Manifest manifest = jar.getManifest();
-            String value = manifest.getMainAttributes().getValue(key);
-            if (value != null) {
+            Optional<String> value = Optional.ofNullable(jar.getManifest())
+                    .map(manifest -> manifest.getMainAttributes().getValue(key));
```

Comment:
It always freaks me out slightly how returning `null` inside a map gets changed into an optional

Reply:
Yeah, also had to double check that it actually does so!

Reply:
It actually means that `map` is really `flatMap` as you can change whether the optional is present or empty. This makes haskell programming inside me die a little.

Reply:
But imagine the bugs you run into otherwise, when your `Optional<T>` is suddenly `null` :P